packages/gatsby/src/commands/build.ts

```diff
@@ -63,6 +64,12 @@ module.exports = async function build(program: IBuildArgs): Promise<void> {
     )
   }
 
+  await createServiceLock(program.directory, `metadata`, {
+    name: program.sitePackageJson.name,
+    sitePath: program.directory,
+    lastRun: Date.now(),
```

Comment:
Is there a reason why this doesn't include `pid`, but the develop one does? I'm _assuming_ this doesn't store the pid since it's not a long-lived process, but just wanted to clarify!

Reply:
Yup, exactly. I only include the pid in case I need to kill the process from Desktop.

Reply:
We can't, because the calling process might not always be the actual build process. We use it in Desktop to write the metadata for sites that we have manually added.